browser/ntp_background/brave_ntp_custom_background_service_delegate.cc

```diff
@@ -123,8 +122,9 @@ bool BraveNTPCustomBackgroundServiceDelegate::HasPreferredBraveBackground()
   }
 
   auto selected_value = pref.GetSelectedValue();
-  if (auto* selected_url = absl::get_if<GURL>(&selected_value)) {
-    return selected_url->is_valid();
+  auto selected_url = GURL(selected_value);
+  if (selected_url != NULL) {
```

Comment:
This doesn't seem right. GURL is a automatic variable, so can't be NULL.

Reply:
And one more thing, in Chromium repository, we usually use C++'s nullptr instead of `NULL`

Reply:
@ajayseeker Sorry for taking your chance to fix this. Just in case where CI get weird, I updated it.

Reply:
> This doesn't seem right. GURL is a automatic variable, so can't be NULL.

Thanks for pointing. Will remember it.

Reply:
> And one more thing, in Chromium repository, we usually use C++'s nullptr instead of `NULL`

sure! will do that from next time.

Reply:
> @ajayseeker Sorry for taking your chance to fix this. Just in case where CI gets weird, I updated it.

It's completely fine, I believe this is only the starting of my contribution journey.

Reply:
Hi @sangwoo108 
I use a M2 MacBook with 8GB of ram and it took me almost a day to build the core. 
I'm curious to know the configuration of system used by you to build and also what is the basic minimum configuration recommended.

Reply:
Also, while we fix the build, Can you please assign me another Issue of your choice?

Reply:
> I use a M2 MacBook with 8GB of ram and it took me almost a day to build the core.

Yeah, it's a very large project, so takes exceptionally long time. I'm using M1 Max with 64GB ram, but takes 3~4 hrs without distributed compilation support. When I use i5-6xxx seriese, it takes almost 10hrs. 😂

Here's a suggestion from Chromium for faster build.

https://chromium.googlesource.com/chromium/src/+/main/docs/mac_build_instructions.md#Faster-builds

You may want to try ccache and `is_debug=false`. `symbol_level=0` is kind of controversial but you can try `blink_symbol_level=0` and `v8_symbol_level=0`. I think you can add this build args to `src/brave/.env` file. But adding these could trigger full build from the scratch again.

---

Awesome! WE'd be grateful if you could help us. Which area are you interested in? like, C++ or javascript, native UI or web ui, and etc. I'll try to find good issues per your preference.

Reply:
I too started the build and went to bed 😃, will try your suggestion for faster build otherwise I don't think with my machine It will be easy. 

I'm interested in C++ and javascript, can try to fix a web ui issue too. Thanks! 🙂

Reply:
Thanks, These are what I couldn't look into.

1. https://github.com/brave/brave-browser/issues/25018
2. https://github.com/brave/brave-browser/issues/26257
3. https://github.com/brave/brave-browser/issues/28873

Maybe, the 3rd item could interest you, which is an experimental feature.

Reply:
sure thank! I'll look into them.